### PR TITLE
virtcontainers: remove temp dir created for vsock in test code

### PR DESCRIPTION
**Pull request by Kvasscn**
remove temp dir generated by mock.GenerateKataMockHybridVSock().

Fixes: #3186

Signed-off-by: zhanghj <zhanghj.lc@inspur.com>

### CODE REVIEW

File: src/runtime/virtcontainers/api_test.go
Comment (Bevisy) [inline src/runtime/virtcontainers/api_test.go:173]: I suggest putting the `defer` operation after the function `hybridVSockTTRPCMock.Start()` where the sock file is actually created.

File: src/runtime/virtcontainers/api_test.go
Comment (Bevisy) [inline src/runtime/virtcontainers/api_test.go:174]: BTW, line 172, the variable `MockHybridVSockPath` here is not required.

Also I found there maybe a mistake in line 176, the param `url` already have prefix "mock://"
```sh
err = hybridVSockTTRPCMock.Start(fmt.Sprintf("mock://%s", url))
-->
err = hybridVSockTTRPCMock.Start(url)
```

File: src/runtime/virtcontainers/api_test.go
Comment (liubin) [inline src/runtime/virtcontainers/api_test.go:173]: Much piecemeal processing spans multiple modules, it is recommended to encapsulate Mock hybrid vsock.

For example, just like this, the mock side:

```golang
type MockHybridVSock struct {
	dir    string
	socket string
}

func (hvs MockHybridVSock) clean() {
	os.RemoveAll(hvs.dir)
}

func (hvs MockHybridVSock) address() string {
	return hvs.socket
}

func GenerateKataMockHybridVSock() (MockHybridVSock, error) {
	dir, err := ioutil.TempDir("", "kata-mock-hybrid-vsock-test")
	if err != nil {
		return MockHybridVSock{}, err
	}

	m := MockHybridVSock{
		dir:    dir,
		socket: fmt.Sprintf(testKataMockHybridVSockURLTempl, dir),
	}

	return m, nil
}
```

And the user side:

```golang
	hvSocket, err := mock.GenerateKataMockHybridVSock()
	assert.NoError(err)
	url := hvSocket.address()
	defer hvSocket.clean()
```

File: src/runtime/virtcontainers/kata_agent_test.go
Comment (Bevisy) [inline src/runtime/virtcontainers/kata_agent_test.go:53]: Same reason for all `defer func()`

**Review by Bevisy — CHANGES_REQUESTED** [review]
Thanks for your PR, but please read my review below.

File: src/runtime/virtcontainers/api_test.go
Comment (Kvasscn) [reply src/runtime/virtcontainers/api_test.go:173]: Thanks for the  advice, and how about adding a function to clean it up,  because  I don't want to change the definition of the  GenerateKataMockHybridVSock() function.  Add a clean function such as:
```
func RemoveKataMockHybridVSock(sockAddress string) error {
    var sockPrefix = "mock://"

    if !strings.HasPrefix(sockAddress, sockPrefix) {
        return fmt.Errorf("Invalid socket address: %s", sockAddress)
    }

    sockPath := strings.TrimPrefix(sockAddress, sockPrefix)

    return os.RemoveAll(path.Dir(sockPath))
}
```
and use  mock. RemoveKataMockHybridVSock(url)  in  the test code.

File: src/runtime/virtcontainers/api_test.go
Comment (Kvasscn) [reply src/runtime/virtcontainers/api_test.go:174]: ok,  i update the commit later.

**Comment by Jakob-Naucke** [comment]
@Kvasscn please rebase :)

**Comment by jodh-intel** [comment]
@Kvasscn - please can you update this branch?

**Comment by jodh-intel** [comment]
Ping @Kvasscn.

**Comment by fidencio** [comment]
A head's up that may affect your PR.

In the Architecture Committee meeting from January 25th, 2022, the Architecture Committee has agreed on using the "Dismiss stale pull request approvals when new commits are pushed" configuration from GitHub.  It basically means that if your PR has been rebased or updated, the approvals given will be erased.

In order to minimize traumas due to the new approach, please, consider adding a note on the changes done before the rebase / force-push, and also pinging the reviewers for a subsequent round of reviews.

Thanks for your understanding!

Related issue: https://github.com/kata-containers/community/issues/249

**Comment by fidencio** [comment]
@Kvasscn, could you rebase this PR atop of `main`?

File: src/runtime/virtcontainers/pkg/mock/mock.go
Comment (jodh-intel) [inline src/runtime/virtcontainers/pkg/mock/mock.go:35]: Nit: It would be safer to make this a top-level `const` and use it to construct `testKataMockHybridVSockURLTempl`

**Review by jodh-intel — APPROVED** [review]
Thanks @Kvasscn.

**Comment by Kvasscn** [comment]
Already rebase and update this PR.  Please review again. @jodh-intel

**Review by jodh-intel — APPROVED** [review]
Thanks @Kvasscn.

lgtm

**Review by liubin — APPROVED** [review]

**Pull request merged**
